astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -1,381 +1,66 @@
+"""
+This module contains operators to interact with Amazon SageMaker in an asynchronous way.
+
+Starting release ``1.17.0`` of the ``astronomer-providers`` package, the operators in this module extend the
+corresponding operators in the Amazon provider of Apache Airflow supporting the deferrable capabilities. With this,
+the aim is to push support and fixes to the upstream operators as much as possible, and also we do not have to keep
+track of the  upstream changes and backport them to the operators in this module.
+"""
+
 from __future__ import annotations
 
-import json
-import time
 from typing import Any
 
-from airflow.exceptions import AirflowException
-from airflow.providers.amazon.aws.hooks.sagemaker import (
-    LogState,
-    secondary_training_status_message,
-)
 from airflow.providers.amazon.aws.operators.sagemaker import (
     SageMakerProcessingOperator,
     SageMakerTrainingOperator,
     SageMakerTransformOperator,
 )
-from airflow.utils.json import AirflowJsonEncoder
-
-from astronomer.providers.amazon.aws.triggers.sagemaker import (
-    SagemakerProcessingTrigger,
-    SagemakerTrainingWithLogTrigger,
-    SagemakerTrigger,
-)
-from astronomer.providers.utils.typing_compat import Context
-
-
-def serialize(result: dict[str, Any]) -> str:
-    """Serialize any objects coming from Sagemaker API response to json string"""
-    return json.loads(json.dumps(result, cls=AirflowJsonEncoder))  # type: ignore[no-any-return]
 
 
 class SageMakerProcessingOperatorAsync(SageMakerProcessingOperator):
```

Comment:
I think we might still need the following for backward compatibility. @phanikumv Could you please confirm? Thanks!

```python
class SageMakerProcessingOperatorAsync(SageMakerProcessingOperator):
        def __init__(self, **kwargs: Any) -> None:
            return super().__init__(deferrable=True, **kwargs)
```

Reply:
I think the idea to extend the upstream providers was to help us not keep us tracking changes in upstream changes and it's backward compatible functionality wise, already no? Our methods din't have different signatures.

Reply:
Yep, but I think this code is for users still using `from astronomer... import SageMakerProcessingOperatorAsync`.

Reply:
Which part exactly you mean? They can still keep doing it. The class is still there.

Reply:
The init method is still there, it accepts both *args and *kwargs for all the operators.

https://github.com/astronomer/astronomer-providers/pull/1169/files#diff-f57678957312458c2669bdee1f0ee1c8a4afe88ea8bf99fef01f69c555b44cddR24

https://github.com/astronomer/astronomer-providers/pull/1169/files#diff-f57678957312458c2669bdee1f0ee1c8a4afe88ea8bf99fef01f69c555b44cddR34

Reply:
Since the diff is huge, maybe it is unclear, 
but this is how the operator module looks like if you checkout to the branch in this PR locally
```
"""
This module contains operators to interact with Amazon SageMaker in an asynchronous way.

Starting release ``1.17.0`` of the ``astronomer-providers`` package, the operators in this module extend the
corresponding operators in the Amazon provider of Apache Airflow supporting the deferrable capabilities. With this,
the aim is to push support and fixes to the upstream operators as much as possible, and also we do not have to keep
track of the  upstream changes and backport them to the operators in this module.
"""

from __future__ import annotations

from typing import Any

from airflow.providers.amazon.aws.operators.sagemaker import (
    SageMakerProcessingOperator,
    SageMakerTrainingOperator,
    SageMakerTransformOperator,
)


class SageMakerProcessingOperatorAsync(SageMakerProcessingOperator):
    """The operator is used to start a SageMaker processing job asynchronously."""

    def __init__(self, *args: Any, **kwargs: Any) -> None:
        """
        Initialize a new SageMakerProcessingOperatorAsync instance.

        We set the deferrable flag to True while calling the super class constructor to enable the deferrable
        capabilities.

        :param args: List of arguments to pass to SageMakerProcessingOperator.
        :param kwargs: Keyword arguments to pass to SageMakerProcessingOperator.
        """
        super().__init__(*args, deferrable=True, **kwargs)


class SageMakerTransformOperatorAsync(SageMakerTransformOperator):
    """The operator is used to start a SageMaker transform job asynchronously."""

    def __init__(self, *args: Any, **kwargs: Any) -> None:
        """
        Initialize a new SageMakerTransformOperatorAsync instance.

        We set the deferrable flag to True while calling the super class constructor to enable the deferrable
        capabilities.

        :param args: List of arguments to pass to SageMakerTransformOperator.
        :param kwargs: Keyword arguments to pass to SageMakerTransformOperator.
        """
        super().__init__(*args, deferrable=True, **kwargs)


class SageMakerTrainingOperatorAsync(SageMakerTrainingOperator):
    """The operator is used to start a SageMaker training job asynchronously."""

    def __init__(self, *args: Any, **kwargs: Any) -> None:
        """
        Initialize a new SageMakerTrainingOperatorAsync instance.

        We set the deferrable flag to True while calling the super class constructor to enable the deferrable
        capabilities.

        :param args: List of arguments to pass to SageMakerTrainingOperator.
        :param kwargs: Keyword arguments to pass to SageMakerTrainingOperator.
        """
        super().__init__(*args, deferrable=True, **kwargs)
```

Reply:
Ah got it. I thought these files are removed 🤦‍♂️ Thanks for your clarification!

Reply:
No, no. The operator module is still there and the init methods should support existing implementation. The GitHub review tab does not show well for large diff when so many lines are getting dropped.